File: ShoeStore/ShoeStore.Domain/Entities/Promotion.cs
namespace ShoeStore.Domain.Entities;

public class Promotion
{
    public int Id { get; set; }
    public string? Code { get; set; }
    public string Name { get; set; } = null!;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int StatusId { get; set; }

    public Status Status { get; set; } = null!;
    public ICollection<PromotionProduct>? PromotionProducts { get; set; }
    public ICollection<PromotionStore>? PromotionStores { get; set; }
}

File: shoestorefe/lib/domain/entities/promotion.dart
import 'promotion_product.dart';
import 'promotion_store.dart';

class Promotion {
  final int id;
  final String? code;
  final String name;
  final DateTime startDate;
  final DateTime endDate;
  final int statusId;
  final String? statusName;
  final List<PromotionProduct> products;
  final List<PromotionStore> stores;

  Promotion({
    required this.id,
    this.code,
    required this.name,
    required this.startDate,
    required this.endDate,
    required this.statusId,
    this.statusName,
    required this.products,
    required this.stores,
  });
}
